AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/AWSCloudWatchLoggingCategoryClient.swift

```diff
@@ -0,0 +1,195 @@
+                controller.enable()
+            }
+        }
+    }
+    
+    func disable() {
+        _enable = false
+        lock.execute {
+            for controller in loggersByKey.values {
+                controller.disable()
+            }
+        }
+    }
+    
+    var `default`: Logger {
+        return self.logger(forCategory: "Amplify")
+    }
+    
+    func logger(forCategory category: String, namespace: String?, logLevel: Amplify.LogLevel) -> Logger {
+        return lock.execute {
+            let key = LoggerKey(category: category, logLevel: logLevel)
+            if let existing = loggersByKey[key] {
+                return existing
+            }
+            
+            
+            let controller = AWSCloudWatchLoggingSessionController(credentialsProvider: credentialsProvider,
+                                                                   authentication: authentication,
+                                                                   logFilter: self.logFilter,
+                                                                   category: category,
+                                                                   namespace: namespace,
+                                                                   logLevel: logLevel,
+                                                                   logGroupName: self.logGroupName,
+                                                                   region: self.region,
+                                                                   localStoreMaxSizeInMB: self.localStoreMaxSizeInMB,
+                                                                   userIdentifier: self.userIdentifier,
+                                                                   networkMonitor: self.networkMonitor)
+            if _enable {
+                controller.enable()
+            }
+            loggersByKey[key] = controller
+            return controller
+        }
+    }
+    
+    func logger(forCategory category: String, logLevel: LogLevel) -> Logger {
+        return self.logger(forCategory: category, namespace: nil, logLevel: logLevel)
+    }
+    
+    func logger(forCategory category: String) -> Logger {
+        let defaultLogLevel = logFilter.getDefaultLogLevel(forCategory: category, userIdentifier: self.userIdentifier)
+        return self.logger(forCategory: category, namespace: nil, logLevel: defaultLogLevel)
+    }
+    
+    func logger(forNamespace namespace: String) -> Logger {
+        self.logger(forCategory: namespace)
+    }
+    
+    func logger(forCategory category: String, forNamespace namespace: String) -> Logger {
+        let defaultLogLevel = logFilter.getDefaultLogLevel(forCategory: category, userIdentifier: self.userIdentifier)
+        return self.logger(forCategory: category, namespace: namespace, logLevel: defaultLogLevel)
+    }
+    
+    func getInternalClient() -> CloudWatchLogsClientProtocol {
+        loggersByKey.first!.value.client!
```

Comment:
Force unwrapping is always risky, but we're doing it two times here. 
How confident are we that it's not gonna result in a crash? Perhaps we could use a `preconditionFailure` instead?

Reply:
There should not be a case where the client is nil if the plugin is configured.  I've updated the logic with a `guard let` check with a `preconditionFailure`